src/aspire/classification/rir_class2d.py

```diff
@@ -184,7 +186,9 @@ def classify(self, diagnostics=False):
 
         # # Stage 2: Compute Nearest Neighbors
         logger.info("Calculate Nearest Neighbors")
-        classes, refl, distances = self.nn_classification(coef_b, coef_b_r)
+        classes, reflections, distances = self.nn_classification(coef_b, coef_b_r)
+        if self.output_nn_filename is not None:
+            self._save_nn(classes, reflections, distances)
```

Comment:
So I don't quite follow when and why we want to do this. When will such a file be read in? Won't this information be available from the API?

Reply:
No worries. I put it in as a convenience for others because I have been asked for it several times now.

It would potentially be used for developing smarter class selection. If a person has a couple basic adj list files saved off from running experiments then they can go off and toy around with making their favorite flavors of Laplacian or network modes etc etc. Figuring out what to do in isolation... If something works, the class2d codes can be altered a little to admit a selector, as we do for an aligner... That would be a nicely scoped project.

Yes the same graph can be extracted numerous ways and places off the top of my head, but it is not specifically exposed, other than as the `classes, reflections, corr` used here in `_save_nn`.  Also, despite this being easy for me, I got the impression other people had trouble with it.  I could go on about the topic, but either we want it or not... I'm not going to add another variant of it for this PR, but I can delete it if you don't want it in here.  If we don't want it here I'll try to do something in a notebook. No good deed goes unpunished :).

Reply:
Ok. I see. How about this, instead of writing to file, couldn't we just have a method that takes in `classes, reflections, corr` (these are output from `classify`, so the user should be able to access them) and outputs the adjacency matrix? Then they can write it to a file or do whatever.

Reply:
I'll take it out for now (shouldn't waste the time...). After the PR is merged I'll make something small adding back such a method. Should be np.